src/axom/primal/geometry/BezierCurve.hpp

```diff
@@ -0,0 +1,335 @@
+   *
+   * \param [in] order the order of the resulting Bezier curve
+   * \pre order is greater than or equal to -1.
+   */
+  explicit BezierCurve(int ord = -1)
+  {
+    SLIC_ASSERT(ord >= -1);
+    const int sz = utilities::max(-1, ord+1);
+    m_controlPoints.resize(sz);
+  }
+
+  /*!
+   * \brief Constructor for an order \a ord= n Bezier curve
+   * from a list of coordinates:
+   * \verbatim {x_0, x_1, x_2,...,x_n,
+   *            y_0, y_1, y_2,...,y_n,
+   *            z_0, z_1, z_2,...,z_n}
+   *
+   * \param [in] pts an array with (n+1)*NDIMS entries, ordered by coordinate
+   * then by polynomial order
+   * \param [in] ord Polynomial order of the curve
+   * \pre order is greater than or equal to zero
+   */
+  BezierCurve(T* pts, int ord)
+  {
+    SLIC_ASSERT(pts != nullptr);
+    SLIC_ASSERT(ord >= 0);
+
+    const int sz = utilities::max(0, ord+1);
+    m_controlPoints.resize(sz);
+
+    for ( int p = 0 ; p <= ord ; p++)
+    {
+      auto& pt = m_controlPoints[p];
+      for ( int j = 0 ; j < NDIMS ; j++)
+      {
+        pt[j]=pts[j*(ord+1)+p];
+      }
+    }
+  }
+
+  /*!
+   * \brief Constructor for a Bezier Curve from an array of coordinates
+   *
+   * \param [in] pts a vector with ord+1 control points
+   * \param [in] ord The Curve's polynomial order
+   * \pre order is greater than or equal to zero
+   *
+   */
+
+  BezierCurve(PointType* pts, int ord)
+  {
+    SLIC_ASSERT(pts != nullptr);
+    SLIC_ASSERT(ord >= 0);
+
+    const int sz = utilities::max(0, ord+1);
+    m_controlPoints.resize(sz);
+
+    for (int p = 0 ; p <= ord ; ++p)
+    {
+      m_controlPoints[p] = pts[p];
+    }
+  }
+
+  /*! Sets the order of the Bezier Curve*/
+  void setOrder( int ord) { m_controlPoints.resize(ord+1); }
+
+  /*! Returns the order of the Bezier Curve*/
+  int getOrder() const { return m_controlPoints.size()-1; }
+
+  /*! Clears the list of control points*/
+  void clear()
+  {
+    const int ord = getOrder();
+    for (int p = 0 ; p <= ord ; ++p)
+    {
+      m_controlPoints[p] = PointType();
+    }
+  }
+
+  /*! Retrieves the control point at index \a idx */
+  PointType& operator[](int idx) { return m_controlPoints[idx]; }
+
+  /*! Retrieves the control point at index \a idx */
+  const PointType& operator[](int idx) const { return m_controlPoints[idx]; }
+
+  /* Checks equality of two Bezier Curve */
+  friend inline bool operator==(const BezierCurve< T, NDIMS>& lhs,
+                                const BezierCurve< T, NDIMS>& rhs)
+  {
+    return lhs.m_controlPoints == rhs.m_controlPoints;
+  }
+
+  friend inline bool operator!=(const BezierCurve< T, NDIMS>& lhs,
+                                const BezierCurve< T, NDIMS>& rhs)
+  {
+    return !(lhs == rhs);
+  }
+
+  /*! Returns a copy of the Bezier curve's control points */
+  CoordsVec getControlPoints() const
+  {
+    return m_controlPoints;
+  }
+
+  /*! Returns an axis-aligned bounding box containing the Bezier curve */
+  BoundingBoxType boundingBox() const
```

Comment:
For consistency, I would rename this method to ``getBoundingBox()``

Reply:
I think we did it this way since ``getBoundingBox()`` seems to imply that the bounding box is precomputed and that the function would simply be returning this, rather than having to compute the bounding box of the control points.

Reply:
Perhaps, `computeBoundingBox()` would be more clear in this case.